aea/package_manager/v1.py

```diff
@@ -179,38 +215,64 @@ def verify(
 
         failed = False
         try:
-            available_packages = self.get_available_package_hashes()
-            for package_id in available_packages:
+            for package_id in self.iter_dependency_tree():
                 self._logger.info(f"Verifying {package_id}")
                 package_path = self.package_path_from_package_id(package_id)
                 configuration_obj = config_loader(
                     package_id.package_type,
                     package_path,
                 )
+                calculated_hash = IPFSHashOnly.get(str(package_path))
 
                 fingerprint_check = check_fingerprint(configuration_obj)
                 if not fingerprint_check:
                     failed = True
-                    self._logger.info(
+                    self._logger.error(
                         f"Fingerprints does not match for {package_id} @ {package_path}"
                     )
+                    if self.is_third_party_package(package_id=package_id):
+                        self._third_party_packages[package_id] = calculated_hash
                     continue
 
                 expected_hash = self.dev_packages.get(
                     package_id, self.third_party_packages.get(package_id)
                 )
-
                 if expected_hash is None:
                     failed = True
-                    self._logger.info(f"Cannot find hash for {package_id}")
+                    self._logger.error(f"Cannot find hash for {package_id}")
                     continue
 
-                calculated_hash = IPFSHashOnly.get(str(package_path))
                 if calculated_hash != expected_hash:
+                    self._logger.error(f"Hash does not match for {package_id}")
+                    self._logger.error(f"\tCalculated hash: {calculated_hash}")
+                    self._logger.error(f"\tExpected hash: {expected_hash}")
+                    failed = True
+                    continue
+
+                dependencies_with_mismatched_hashes = self.check_dependencies(
```

Comment:
similar code block exists above

Reply:
That's for package manager v0 which we will be deprecating soon, this is for v1